knowledge_repo/app/models.py

```diff
@@ -350,8 +350,18 @@ def groups(self, groups):
 
         # create an implicit group, group_post.id, to add
         # single users to
-        group = Group(name="group_" + self.id)
+        # we don't have self.id here necessarily, so query the Post table
```

Comment:
Hi Nikki! Under which circumstances do we not have the id? If the id can be obtained using the query below, then it should also be available via self.id, no? What am I missing?